Error: Failed to find nav item

v1.0.4 - after the extension install or sometimes after reload of the page
Error logged: Error: Failed to find nav item
    at c (inboxsdk.js:143:198625)
    at n.default (inboxsdk.js:143:160009) 
Original error stack:
Error: Failed to find nav item
    at c (chrome-extension://kibfopgnjcjgblnojndmnjcdnmaghedd/js/utils/inboxsdk.js:143:198625)
    at n.default (chrome-extension://kibfopgnjcjgblnojndmnjcdnmaghedd/js/utils/inboxsdk.js:143:160009) 
Error logged from:
    at n.default (chrome-extension://kibfopgnjcjgblnojndmnjcdnmaghedd/js/utils/inboxsdk.js:143:56113)
    at I (chrome-extension://kibfopgnjcjgblnojndmnjcdnmaghedd/js/utils/inboxsdk.js:252:7021)
    at S.error (chrome-extension://kibfopgnjcjgblnojndmnjcdnmaghedd/js/utils/inboxsdk.js:252:5826)
    at c (chrome-extension://kibfopgnjcjgblnojndmnjcdnmaghedd/js/utils/inboxsdk.js:143:198619)
    at n.default (chrome-extension://kibfopgnjcjgblnojndmnjcdnmaghedd/js/utils/inboxsdk.js:143:160009) 

![image](https://user-images.githubusercontent.com/25505945/228599356-0e368a3e-7dc6-4dd6-9972-a4eb09cb1140.png)

I managed to trigger this error with [an example extension](https://github.com/InboxSDK/InboxSDK/blob/64bff328fecb9d5854677c89fa6c33fc1c38c6ae/examples/compose-tooltip-button/content.js#L4)

Stack trace with source maps

```
inboxsdk.js:1735 Error logged: Error: Failed to find nav item
    at isDarkTheme (inboxsdk.js:5622:63)
    at gmailLoadEvent (inboxsdk.js:3833:82) 

Original error stack:
Error: Failed to find nav item
    at isDarkTheme (chrome-extension://cdocalkpnnafjienffbkmlgfdokghmpe/inboxsdk.js:5622:63)
    at gmailLoadEvent (chrome-extension://cdocalkpnnafjienffbkmlgfdokghmpe/inboxsdk.js:3833:82) 

Error logged from:
    at logError (chrome-extension://cdocalkpnnafjienffbkmlgfdokghmpe/inboxsdk.js:1719:82)
    at _logError (chrome-extension://cdocalkpnnafjienffbkmlgfdokghmpe/inboxsdk.js:23605:64)
    at Logger.error (chrome-extension://cdocalkpnnafjienffbkmlgfdokghmpe/inboxsdk.js:23388:5)
    at isDarkTheme (chrome-extension://cdocalkpnnafjienffbkmlgfdokghmpe/inboxsdk.js:5622:57)
    at gmailLoadEvent (chrome-extension://cdocalkpnnafjienffbkmlgfdokghmpe/inboxsdk.js:3833:82) 

Extension App Ids: [
  {
    "appId": "simple-example"
  }
] 
Sent by App: false 
Session Id: 1691000019023-0.40742939578700454 
Extension Id: chrome-extension://cdocalkpnnafjienffbkmlgfdokghmpe/ 
InboxSDK Loader Version: 0.7.25-1690914166116-1ccd156ad64a143a-MODIFIED-WATCH 
InboxSDK Implementation Version: 0.7.25-1690914166116-1ccd156ad64a143a-MODIFIED-WATCH 
Is Using Sync API: true
```

I think the fix might just involve waiting for nav menu item selectors to come up like we do in [here](https://github.com/InboxSDK/InboxSDK/blob/64bff328fecb9d5854677c89fa6c33fc1c38c6ae/src/platform-implementation-js/dom-driver/gmail/gmail-driver/add-nav-item.ts#L45) for other parts of the `gmailLoadEvent`.

--- a/src/platform-implementation-js/dom-driver/gmail/gmail-driver/gmail-load-event.ts
+++ b/src/platform-implementation-js/dom-driver/gmail/gmail-driver/gmail-load-event.ts
@@ -3,7 +3,7 @@ import isIntegratedViewGmail, {
   isCollapsiblePanelHidden,
 } from './isIntegratedViewGmail';
 import { waitForMenuReady } from './add-nav-item';
-import { isDarkTheme } from './track-gmail-styles';
+import { checkForDarkThemeSafe } from './track-gmail-styles';
 
 export default async function gmailLoadEvent(driver: GmailDriver) {
   const pageCommunicator = driver.getPageCommunicator();
@@ -11,11 +11,12 @@ export default async function gmailLoadEvent(driver: GmailDriver) {
   const isConversationViewDisabled =
     await pageCommunicator.isConversationViewDisabled();
   await waitForMenuReady();
+  const isUsingDarkTheme = await checkForDarkThemeSafe();
 
   driver.getLogger().eventSite('gmailSettings', {
     isGmailIntegratedView: isIntegratedViewGmail(),
     isCollapsiblePanelHidden: isCollapsiblePanelHidden(),
-    isDarkTheme: isDarkTheme(),
+    isDarkTheme: isUsingDarkTheme,
     screenWidth: window.screen?.width,
     screenHeight: window.screen?.height,
     windowWidth: window.innerWidth,
@@ -27,24 +28,22 @@ export default async function gmailLoadEvent(driver: GmailDriver) {
     wasAccountSwitcherReadyAtStart:
       driver.getEnvData().wasAccountSwitcherReadyAtStart,
     timings: {
-      ...{
-        responseStart: global.performance?.timing.responseStart,
-        responseEnd: global.performance?.timing.responseEnd,
-        domLoading: global.performance?.timing.domLoading,
-        domInteractive: global.performance?.timing.domInteractive,
-        domComplete: global.performance?.timing.domComplete,
-        domContentLoadedEventStart:
-          global.performance?.timing.domContentLoadedEventStart,
-        domContentLoadedEventEnd:
-          global.performance?.timing.domContentLoadedEventEnd,
-        domainLookupEnd: global.performance?.timing.domainLookupEnd,
-        domainLookupStart: global.performance?.timing.domainLookupStart,
-        fetchStart: global.performance?.timing.fetchStart,
-        loadEventStart: global.performance?.timing.loadEventStart,
-        loadEventEnd: global.performance?.timing.loadEventEnd,
-        navigationStart: global.performance?.timing.navigationStart,
-        requestStart: global.performance?.timing.requestStart,
-      },
+      responseStart: global.performance?.timing.responseStart,
+      responseEnd: global.performance?.timing.responseEnd,
+      domLoading: global.performance?.timing.domLoading,
+      domInteractive: global.performance?.timing.domInteractive,
+      domComplete: global.performance?.timing.domComplete,
+      domContentLoadedEventStart:
+        global.performance?.timing.domContentLoadedEventStart,
+      domContentLoadedEventEnd:
+        global.performance?.timing.domContentLoadedEventEnd,
+      domainLookupEnd: global.performance?.timing.domainLookupEnd,
+      domainLookupStart: global.performance?.timing.domainLookupStart,
+      fetchStart: global.performance?.timing.fetchStart,
+      loadEventStart: global.performance?.timing.loadEventStart,
+      loadEventEnd: global.performance?.timing.loadEventEnd,
+      navigationStart: global.performance?.timing.navigationStart,
+      requestStart: global.performance?.timing.requestStart,
       ...driver.getTimings(),
     },
   });

--- a/src/platform-implementation-js/dom-driver/gmail/gmail-driver/track-gmail-styles.ts
+++ b/src/platform-implementation-js/dom-driver/gmail/gmail-driver/track-gmail-styles.ts
@@ -1,6 +1,6 @@
 import kefirBus from 'kefir-bus';
 import Logger from '../../../lib/logger';
-import waitFor from '../../../lib/wait-for';
+import waitFor, { WaitForError } from '../../../lib/wait-for';
 import GmailElementGetter from '../gmail-element-getter';
 
 const RGB_REGEX = /^rgb\s*\(\s*(\d+),\s*(\d+),\s*(\d+)\s*\)/;
@@ -18,10 +18,31 @@ function getDensity(): 'compact' | 'default' {
 
 const navItemSelector = '.aio' as const;
 
-export function isDarkTheme(): boolean {
+function getNavItem() {
+  return document.querySelector(navItemSelector);
+}
+
+/**
+ * @returns true if Gmail is in dark theme mode, false if not, or null if it can't be determined
+ */
+export async function checkForDarkThemeSafe() {
+  try {
+    await waitFor(() => getNavItem());
+  } catch (e: unknown) {
+    if (e instanceof WaitForError) {
+      return null;
+    }
+
+    throw e;
+  }
+
+  return isDarkTheme();
+}
+
+function isDarkTheme(): boolean {
   // get the color of the left-nav-menu entries to determine whether Gmail is
   // in dark theme mode.
-  const navItem = document.querySelector(navItemSelector);
+  const navItem = getNavItem();
   if (!navItem) {
     Logger.error(new Error('Failed to find nav item'));
     return false;

--- a/src/platform-implementation-js/lib/wait-for.ts
+++ b/src/platform-implementation-js/lib/wait-for.ts
@@ -1,10 +1,21 @@
+export class WaitForError extends Error {
+  name = 'WaitForError';
+  constructor() {
+    super('waitFor timeout');
+  }
+}
+
+/**
+ * @param condition a function that returns the value to wait for, or falsey if it's not ready yet
+ * @throws {WaitForError} if the condition is not met within the timeout
+ */
 export default function waitFor<T>(
   condition: () => T | null | undefined,
   timeout: number = 120 * 1000,
   steptime: number = 250
 ): Promise<NonNullable<T>> {
   // make this error here so we have a sensible stack.
-  const timeoutError = new Error('waitFor timeout');
+  const timeoutError = new WaitForError();
 
   return new Promise(function (resolve, reject) {
     let waited = 0;

